server/channels/api4/outgoing_oauth_connection.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package api4
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/v8/einterfaces"
+)
+
+const (
+	whereOutgoingOAuthConnection = "outgoingOAuthConnections"
+)
+
+func (api *API) InitOutgoingOAuthConnection() {
+	api.BaseRoutes.OutgoingOAuthConnections.Handle("", api.APISessionRequired(listOutgoingOAuthConnections)).Methods("GET")
+	api.BaseRoutes.OutgoingOAuthConnection.Handle("", api.APISessionRequired(getOutgoingOAuthConnection)).Methods("GET")
+}
+
+func ensureOutgoingOAuthConnectionInterface(c *Context, where string) (einterfaces.OutgoingOAuthConnectionInterface, bool) {
+	if !c.App.Config().FeatureFlags.OutgoingOAuthConnections {
+		c.Err = model.NewAppError(where, "api.context.outgoing_oauth_conneciton.not_available.feature_flag", nil, "", http.StatusNotImplemented)
+		return nil, false
+	}
+
+	if c.App.OutgoingOAuthConnection() == nil || c.App.License() == nil || c.App.License().SkuShortName != model.LicenseShortSkuEnterprise {
```

Comment:
`c.App.OutgoingOAuthConnection()` makes me think there is one Outgoing OAuth Connection supported by the server, and it's not clear what that "one" connection is meant to connect to